AK/Format.h

```diff
@@ -693,6 +694,14 @@ struct Formatter<ErrorOr<T, ErrorType>> : Formatter<FormatString> {
     }
 };
 
+template<>
+struct Formatter<SourceLocation> : Formatter<FormatString> {
```

Comment:
What is the logic for moving this? (This should really be in the commit description please)

`Formatter` implementations generally live with the thing their formatting, not in the format library itself?

Reply:
Yeah, I'm sorry. It was causing a dependency issue since SourceLocation.h was including Error.h and Error.h SourceLocation.h